applications/tari_indexer/src/substate_manager.rs

```diff
@@ -251,6 +255,23 @@ impl SubstateManager {
         Ok(nfts)
     }
 
+    pub async fn get_event_from_db(
+        &self,
+        template_address: String,
+        tx_hash: String,
+    ) -> Result<Vec<EventData>, anyhow::Error> {
+        let mut tx = self.substate_store.create_read_tx()?;
+        let events = tx.get_events(template_address, tx_hash)?;
+        Ok(events)
+    }
+
+    pub async fn save_event_to_db(&self, new_event: NewEvent) -> Result<(), anyhow::Error> {
```

Comment:
Same for `NewEvent` (diesel struct). For the payload, we dont want the caller to choose how to serialize something. If we're defining payload as a HashMap<String, String> then use that type. 

```suggestion
    pub async fn save_event_to_db(&self, template_address: TemplateAddress, tx_hash: PayloadId, topic: String, payload: &HashMap<String, String>) -> Result<(), anyhow::Error> {
// or even
    pub async fn save_event_to_db<T: Serialize>(&self, template_address: TemplateAddress, tx_hash: PayloadId, topic: String, payload: &T) -> Result<(), anyhow::Error> {
// where in the current implementation T happens to be serialized to JSON
```

Reply:
That's slightly more subtle, as I think GraphQL doesn't support, at the moment of writing and if I am not mistaken, queries parametrized by `HashMap` directly.

Reply:
Was more referring to the database layer, not graphQL, but I see you came right with the changes :) There is no PR description, but I'm curious what the use case is for saveEvent in the graphQL interface. Happy to merge as is though